features/functests/sctp/sctp.go

```diff
@@ -36,56 +38,104 @@ var _ = Describe("sctp", func() {
 		namespace.Create(testNamespace, clients.K8s)
 		namespace.Clean(testNamespace, clients.K8s)
 		checkForSctpReady(clients.K8s)
-		By("Creating the sctp service")
-		createSctpService(clients.K8s)
 	})
-
-	var _ = Context("Client Server Connection", func() {
+	var _ = Describe("sctp setup", func() {
 		var clientNode string
 		var serverNode string
-		It("Should connect a client pod to a server pod", func() {
-			By("Choosing the nodes for the server and the client")
-			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
-				LabelSelector: "node-role.kubernetes.io/worker=",
-			})
-			Expect(err).ToNot(HaveOccurred())
-			clientNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
-			serverNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
-			if len(nodes.Items) > 1 {
-				serverNode = nodes.Items[1].ObjectMeta.Labels[hostnameLabel]
-			}
+		var activeService *k8sv1.Service
+		var runningPod *k8sv1.Pod
+		var serverPod *k8sv1.Pod
+		var nodes *k8sv1.NodeList
+		var err error
 
-			By("Starting the server")
-			serverArgs := []string{"-ip", "0.0.0.0", "-port", "30101", "-server"}
-			pod := scptTestPod("scptserver", serverNode, "sctpserver", serverArgs)
-			pod.Spec.Containers[0].Ports = []k8sv1.ContainerPort{
-				k8sv1.ContainerPort{
-					Name:          "sctpport",
-					Protocol:      k8sv1.ProtocolSCTP,
-					ContainerPort: 30101,
-				},
-			}
-			serverPod, err := clients.K8s.CoreV1().Pods(testNamespace).Create(pod)
-			Expect(err).ToNot(HaveOccurred())
+		// OCP-26759
+		Context("Test Kerenel Module", func() {
+			FIt("Kernel Module is loaded", func() {
+				checkForSctpReady(clients.K8s)
+			})
+		})
 
-			By("Fetching the server's ip address")
-			var runningPod *k8sv1.Pod
-			Eventually(func() k8sv1.PodPhase {
-				runningPod, err = clients.K8s.CoreV1().Pods(testNamespace).Get(serverPod.Name, metav1.GetOptions{})
+		var _ = Describe("Test Connectivity", func() {
+			BeforeEach(func() {
+				By("Choosing the nodes for the server and the client")
+				nodes, err = clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+					LabelSelector: "node-role.kubernetes.io/worker-sctp=",
```

Comment:
We may end up with a worker-A / worker-B machine pools, so this label is likely to change.

Reply:
Let's do the other way around, labelling only some nodes as worker-sctp